packages/flutter/lib/src/cupertino/segment_control.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/painting.dart';
+
+class SegmentControlItem {
+  final String title;
+  final Widget content;
+
+  SegmentControlItem(this.title, this.content);
+}
+
+class SegmentControl extends StatefulWidget {
+  final List<SegmentControlItem> tabs;
+  final int activeTabIndex;
+
+  SegmentControl(this.tabs, {this.activeTabIndex = 0})
+      : assert(tabs.length > 1 && tabs.length <= 3), assert(activeTabIndex <= tabs.length - 1);
+
+  @override
+  State createState() {
+    return new _SegmentControlState();
+  }
+}
+
+class _SegmentControlState extends State<SegmentControl> {
+  int _activeTabIndex;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _activeTabIndex = widget.activeTabIndex;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget activeTab = widget.tabs[_activeTabIndex].content;
+
+    List<_SegmentControlItem> list = <_SegmentControlItem>[];
+
+    for (int i = 0; i < widget.tabs.length; i++) {
+      SegmentControlItem tap = widget.tabs[i];
+      bool isActive = tap == widget.tabs[_activeTabIndex];
+      _ButtonPlace place = _ButtonPlace.start;
+
+      if (i > 0 && (widget.tabs.length - 1 == i)) {
+        place = _ButtonPlace.end;
+      } else if (i > 0 && (widget.tabs.length - 1 > i)) {
+        place = _ButtonPlace.middle;
+      }
+
+      list.add(new _SegmentControlItem(this, tap, place, isActive));
+    }
+
+    return new Column(
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: <Widget>[
+        new Padding(
+          child: new Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: list,
+          ),
+          padding: new EdgeInsets.all(12.0),
+        ),
+        activeTab
```

Comment:
add trailing commas

Reply:
ahhh I see, you've bundled the segmented control and the view each segment switches to together. 

The Flutter framework strongly prefers the usage of many small composable components https://flutter.io/technical-overview rather than all-in-one widgets. 

Preferably this should be split into just the controls bar which just calls out to a callback which segment is selected (the meat of the code here I think) and a separate entity for responding to switching tabs. 

That also has the added benefit of not having to fully build out the initially invisible tabs that may never be navigated to (and that may need network queries to construct in the first place etc etc).